service/src/main/java/org/grnet/cat/services/CommentService.java

```diff
@@ -0,0 +1,100 @@
+package org.grnet.cat.services;
+
+import jakarta.enterprise.context.ApplicationScoped;
+import jakarta.inject.Inject;
+import jakarta.transaction.Transactional;
+import jakarta.ws.rs.ForbiddenException;
+import jakarta.ws.rs.core.UriInfo;
+import lombok.Value;
+import org.grnet.cat.dtos.assessment.CommentRequestDto;
+import org.grnet.cat.dtos.assessment.CommentResponseDto;
+import org.grnet.cat.dtos.guidance.GuidanceResponseDto;
+import org.grnet.cat.dtos.pagination.PageResource;
+import org.grnet.cat.entities.Assessment;
+import org.grnet.cat.entities.Comment;
+import org.grnet.cat.mappers.CommentMapper;
+import org.grnet.cat.mappers.GuidanceMapper;
+import org.grnet.cat.repositories.AssessmentRepository;
+import org.grnet.cat.repositories.CommentRepository;
+import org.grnet.cat.repositories.UserRepository;
+import org.jboss.logging.Logger;
+
+import java.time.LocalDateTime;
+
+@ApplicationScoped
+public class CommentService {
+
+    @Inject
+    CommentRepository commentRepository;
+
+    @Inject
+    AssessmentRepository assessmentRepository;
+
+    @Inject
+    UserRepository userRepository;
+
+
+
+    private static final Logger LOG = Logger.getLogger(CommentService.class);
+
+    /**
+     * Adds a comment to the specified assessment.
+     *
+     * @param assessmentId The ID of the assessment.
+     * @param commentRequestDto      The comment to add.
+     * @return The added comment.
+     */
+    @Transactional
+    public CommentResponseDto addCommentToAssessment(String assessmentId, String userId, CommentRequestDto commentRequestDto) {
+
+        var assessment = assessmentRepository.findById(assessmentId);
+        var user = userRepository.fetchUser(userId);
+        var comment = CommentMapper.INSTANCE.commentRequestToEntity(commentRequestDto);
+
+        comment.setAssessment(assessment);
+        comment.setUser(user);
+        comment.setText(commentRequestDto.text);
+        comment.setCreatedOn(LocalDateTime.now());
+
+        commentRepository.persist(comment);
+
+        assessment.getComments().add(comment);
+        assessmentRepository.persist(assessment);
```

Comment:
Remove these two lines of code. The comment is being saved from the persist above.